### ISSUE TITLE
Eval bug: Crash in `llama-completion` using MinimaxMs

### ISSUE THREAD

**Issue by am17an** [ggml-org/llama.cpp#18715]
### Name and Version

Using `MiniMax-M2-q5_k_m.gguf` I get this crash using `llama-completion` for a couple of tokens.

```
#0  0x00007f4bf449f002 in ?? () from /usr/lib/libc.so.6
#1  0x00007f4bf449316c in ?? () from /usr/lib/libc.so.6
#2  0x00007f4bf44931b4 in ?? () from /usr/lib/libc.so.6
#3  0x00007f4bf4503d8f in wait4 () from /usr/lib/libc.so.6
#4  0x00007f4c0296ab6b in ggml_print_backtrace () from /home/amang/experiments/llama.cpp/build_n/bin/libggml-base.so.0
#5  0x00007f4c0297d4d9 in ggml_uncaught_exception() () from /home/amang/experiments/llama.cpp/build_n/bin/libggml-base.so.0
#6  0x00007f4bf48b1eba in __cxxabiv1::__terminate (handler=<optimized out>) at /usr/src/debug/gcc/gcc/libstdc++-v3/libsupc++/eh_terminate.cc:48
⚠️ warning: 48  /usr/src/debug/gcc/gcc/libstdc++-v3/libsupc++/eh_terminate.cc: No such file or directory
#7  0x00007f4bf48975d9 in std::terminate () at /usr/src/debug/gcc/gcc/libstdc++-v3/libsupc++/eh_terminate.cc:58
58      in /usr/src/debug/gcc/gcc/libstdc++-v3/libsupc++/eh_terminate.cc
#8  0x00007f4bf48b2176 in __cxxabiv1::__cxa_throw (obj=<optimized out>, tinfo=0x7f4bf4a84d78 <typeinfo for std::out_of_range>, dest=0x7f4bf48c9d00 <std::out_of_range::~out_of_range()>) at /usr/src/debug/gcc/gcc/libstdc++-v3/libsupc++/eh_throw.cc:98
⚠️ warning: 98  /usr/src/debug/gcc/gcc/libstdc++-v3/libsupc++/eh_throw.cc: No such file or directory
#9  0x00007f4bf489bf2a in std::__throw_out_of_range_fmt (__fmt=<optimized out>) at /usr/src/debug/gcc/gcc/libstdc++-v3/src/c++11/functexcept.cc:101
⚠️ warning: 101 /usr/src/debug/gcc/gcc/libstdc++-v3/src/c++11/functexcept.cc: No such file or directory
#10 0x000055e8752cd39c in common_chat_format_single[abi:cxx11](common_chat_templates const*, std::vector<common_chat_msg, std::allocator<common_chat_msg> > const&, common_chat_msg const&, bool, bool) [clone .cold] ()
#11 0x000055e8752f4e6e in main::{lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}::operator()(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const ()
#12 0x000055e8752f18e0 in main ()
[Inferior 1 (process 230508) detached]
terminate called after throwing an instance of 'std::out_of_range'
  what():  basic_string::substr: __pos (which is 461) > this->size() (which is 147)
```

### Operating systems

Linux

### GGML backends

CUDA

### Hardware

4x 4090, 1x 5090 

### Problem description & steps to reproduce

`build_n/bin/llama-completion -m /opt//models/MiniMax-M2-iq3_xs.gguf --cpu-moe --jinja`

### Relevant log output

<details>
<summary>Logs</summary>

```console

```
</details>

**Comment by CISC** [comment]
`llama-completion` doesn't handle `jinja` templates that change message history.

**Comment by am17an** [comment]
Hmm, I only used `--jinja` cos without it the crash is different with the error message `  what():  this custom template is not supported, try using --jinja`.

**Comment by CISC** [comment]
There's a "fix" for it in #16603 but ultimately `llama-completion` won't work correctly with such templates, still, might be worth extracting it to avoid the crash at least.

**Comment by ngxson** [comment]
If the chat template is not supported, you can manually take one from --chat-template. Or better, format the prompt manually and use -no-cnv option for raw completion.

The support for chat template in llama-completion has always been best-effort. Indeed, the default chat option in llama-completion show now be retired and it should now be a raw completion tool by default.

**Comment by am17an** [comment]
I just use `llama-completion` as a quick smoke test to test the sanity of my changes. I guess I can just move over to `llama-cli` for now. I'm not sure if there are any drawbacks

**Comment by ngxson** [comment]
Given your work is mostly on backend side, I think raw completion `llama-completion -no-cnv -p "I believe the meaning of life is"` should fit better for your use case.

I don't recommend using any chat templates when dealing with backend changes because some template has current date in system prompt that can change dynamically. These kind of dynamic chat template can affect the logits and that can affect test results.
